specs/NavigateWithWebResourceRequest.md

```diff
@@ -0,0 +1,143 @@
+    CHECK_FAILURE(webview->NavigateWithWebResourceRequest(webResourceRequest.get()));
+```
+
+```csharp
+    UTF8Encoding utfEncoding = new UTF8Encoding();
+    byte[] postData = utfEncoding.GetBytes("input=Hello");
+
+    using (MemoryStream postDataStream = new MemoryStream(postData.Length))
+    {
+      postDataStream.Write(postData, 0, postData.Length);
+      CoreWebView2WebResourceRequest webResourceRequest = 
+        environment.CreateWebResourceRequest(
+          "https://www.w3schools.com/action_page.php",
+          "POST",
+          postDataStream,
+          "Content-Type: application/x-www-form-urlencoded");
+      webView.CoreWebView2.NavigateWithWebResourceRequest(webResourceRequest);
+    }
+```
+
+# Remarks
+<!-- Explanation and guidance that doesn't fit into the Examples section. -->
+
+<!-- APIs should only throw exceptions in exceptional conditions; basically,
+only when there's a bug in the caller, such as argument exception.  But if for some
+reason it's necessary for a caller to catch an exception from an API, call that
+out with an explanation either here or in the Examples -->
+
+# API Notes
+<!-- Option 1: Give a one or two line description of each API (type
+and member), or at least the ones that aren't obvious
+from their name.  These descriptions are what show up
+in IntelliSense. For properties, specify the default value of the property if it
+isn't the type's default (for example an int-typed property that doesn't default to zero.) -->
+
+<!-- Option 2: Put these descriptions in the below API Details section,
+with a "///" comment above the member or type. -->
+
+# API Details
+<!-- The exact API, in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/) -->
+```idl
+[uuid(7fbad153-fb94-452e-acab-3cbb9ab341ec), object, pointer_default(unique)]
+interface ICoreWebView2_2 : ICoreWebView2 {
+ ....
+  /// Navigate using a constructed WebResourceRequest object. This let's you
+  /// provide post data or additional request headers during navigation.
+  /// The headers in the WebResourceRequest override headers
+  /// added by WebView2 runtime except for Cookie headers. To override cookies
+  /// please use the Cookie API.
+  /// Method can only be either "GET" or "POST". Provided post data will only
+  /// be sent only if the method is "POST" and the uri scheme is HTTP(S).
+  /// \snippet ScenarioNavigateWithWebResourceRequest.cpp NavigateWithWebResourceRequest
+  HRESULT NavigateWithWebResourceRequest([in] ICoreWebView2WebResourceRequest* request);
+}
+
+[uuid(1c11735a-d57d-4614-a9e1-8b48d81da38c), object, pointer_default(unique)]
+interface ICoreWebView2Environment_2 : ICoreWebView2Environment {
+  /// Create a new web resource request object.
+  /// URI parameter must be absolute URI.
+  /// The headers string is the raw request header string delimited by newline
+  /// (\r\n). It's also possible to create this object with null headers string
```

Comment:
But it seems that the final header is *not* followed by `\r\n`.

Reply:
That's not required for last header. Is that something we should specify here.

Reply:
I was thinking of this as a string that is http headers delimited by \r\n which wouldn't have a terminating \r\n, but the raw HTTP request headers as a string would end with a terminating \r\n. It sounds like you're implying its optional. If so, an update to the docs to note that a terminating \r\n at the end of the headers string is optional.

Reply:
Please be clear whether the trailing `\r\n` is (1) optional, or (2) forbidden.